pkg/util/kubernetes/auth.go

```diff
@@ -6,22 +6,54 @@
 package kubernetes
 
 import (
+	"crypto/tls"
+	"crypto/x509"
+	"fmt"
 	"io/ioutil"
-
-	log "github.com/cihub/seelog"
+	"os"
 )
 
-// Kubelet constants
+// Kubernetes constants
 const (
-	AuthTokenPath = "/var/run/secrets/kubernetes.io/serviceaccount/token"
+	ServiceAccountPath      = "/var/run/secrets/kubernetes.io/serviceaccount"
+	ServiceAccountTokenPath = ServiceAccountPath + "/token"
 )
 
-// GetAuthToken reads the serviceaccount token
-func GetAuthToken() string {
-	token, err := ioutil.ReadFile(AuthTokenPath)
+// IsServiceAccountToken returns if a service account token is present on the const path
+func IsServiceAccountToken() bool {
+	_, err := os.Stat(ServiceAccountTokenPath)
+	return err == nil
+}
+
+// GetBearerToken reads the serviceaccount token
+func GetBearerToken(authTokenPath string) (string, error) {
+	token, err := ioutil.ReadFile(authTokenPath)
+	if err != nil {
+		return "", fmt.Errorf("could not read token from %s: %s", authTokenPath, err)
+	}
+	return fmt.Sprintf("bearer %s", token), nil
+}
+
+// GetCertificates loads the certificate and the private key
+func GetCertificates(certFilePath, keyFilePath string) ([]tls.Certificate, error) {
```

Comment:
for clarity, could we call it `GetClientCertificates`?

Reply:
It's not for the clients only as we planed to use it to serveTLS in the DCA-API.